### ISSUE TITLE
Threat Modeling - wallet_invokeSnaps - Information Disclosure

### ISSUE THREAD

**Issue by bentobox19** [MetaMask/snaps-monorepo#1145]
Error messages may need to be reviewed.

An attacker can abuse the method `wallet_invokeSnaps` to obtain information about the installed snaps.

**Comment by FrederikBolding** [comment]
I don't think this is the case. If a DApp tries to use `wallet_invokeSnap` on a snap it doesn't have permission to use it will get the following error:
```
Unauthorized to perform action. Try requesting the required permission(s) first. For more information, see: https://docs.metamask.io/guide/rpc-api.html#permissions
```

This error will not change even if the snap is already installed but not available to the DApp.
